public/app/plugins/datasource/loki/querybuilder/operationUtils.ts

```diff
@@ -257,6 +257,12 @@ export function addNestedQueryHandler(def: QueryBuilderOperationDef, query: Loki
 
 export function getLineFilterRenderer(operation: string) {
   return function lineFilterRenderer(model: QueryBuilderOperation, def: QueryBuilderOperationDef, innerExpr: string) {
+    if (
+      model.id === LokiOperationId.LineContainsInsensitive ||
+      model.id === LokiOperationId.LineContainsNotInsensitive
+    ) {
+      return `${innerExpr} ${operation} \`(?i)${model.params[0]}\``;
+    }
     return `${innerExpr} ${operation} \`${model.params[0]}\``;
   };
 }
```

Comment:
I'm inclined to suggest the following alternative implementation:

```suggestion
export function getLineFilterRenderer(operation: string, caseInsensitive?: boolean = false) {
  return function lineFilterRenderer(model: QueryBuilderOperation, def: QueryBuilderOperationDef, innerExpr: string) {
    if (caseInsensitive) {
      return `${innerExpr} ${operation} \`(?i)${model.params[0]}\``;
    }
    return `${innerExpr} ${operation} \`${model.params[0]}\``;
  };
}
```

Reply:
@matyax much better solution, i'll implement this now.